File: vol11/1100.cpp
#include <iostream>
#include <cstdio>
using namespace std;

int main()
{
    int n;
    int count = 0;

    while (cin >> n) {
        if (n == 0) {
            break;
        }

        int x[n], y[n];

        for (int i = 0; i < n; ++i) {
            cin >> x[i] >> y[i];
        }

        float pol = 0;

        for (int k = 0; k < n - 1; k++) {
            pol += x[k] * y[k + 1] - x[k + 1] * y[k];
        }

        pol += x[n - 1] * y[0] - x[0] * y[n - 1];

        if (pol < 0) {
            pol *= -1;
        }

        printf("%d %#.1f\n", ++count, pol / 2);
    }
}

File: vol11/1100.py
# -*- coding: utf-8 -*-

count = 1
while True:
    n = int(input())
    if n == 0:
        break

    x = []
    y = []
    for i in range(n):
        t = list(map(int, input().split()))
        x.append(t[0])
        y.append(t[1])

    pol = 0.0
    for j in range(n - 1):
        pol += x[j] * y[j + 1] - x[j + 1] * y[j]

    pol += x[n - 1] * y[0] - x[0] * y[n - 1]

    if pol < 0:
        pol *= -1

    print("%d %.1f" % (count, pol / 2))
    count += 1
    input()  # 入れないとおちる
